public/javascript/pump/view.js

```diff
@@ -725,6 +725,11 @@
             view.setupSubs();
             // Initialize state of login button
             view.onKey();
+
+            // Waiting to the view is completely ready before focus the username field
+            setTimeout(function() {
```

Comment:
Why do we have to have `setTimeout()` here? AFAICT by the time this code runs the view _should_ be completely ready.

Reply:
Backbone prevents the .focus immediately, is the same error that happens with `autofocus`, the DOM  is ready but, not the view

Reply:
Maybe I'm missing something obvious, in which case I apologize, but I'm still not convinced that's correct. That `ready` handler is on the Backbone view, not the document - AFAICT the event being handled _is_ "the view is ready". That's why `view.onKey()` works. Plus (again AFAICT) `view.setupSubs()` is synchronous so everything _should_ be ready. What am I missing?

Reply:
the view gets re-rendered because of the collection/model changes once after the view is ready and re-rendering loses focus from any input. you can try your self putting only `view.$('#login input[name="nickname"]').focus();`

Reply:
Can we listen for the view being re-rendered then?